In [49]:
import importlib

import analysis_utils

### Growth of GF2 Matrices over increasing n's
[To Paper](https://www.emis.de/journals/SC/1997/2/pdf/smf_sem-cong_2_35-42.pdf)

To Calculate even n's
$$ 
C_{2n}(q) = \sum^{n/3}_{j=-(n+1)/3} [\binom{2n}{n-3j} - \binom{2n}{n-1-3j}] * q^{n^{2} - 3j^{2} - j}
$$
To simplify this expression I set n = floor(n/2)


To Calculate odd n's
$$
C_{2n + 1}(q) = \sum^{n/3}_{j=-(n+2)/3} [\binom{2n + 1}{n-3j} - \binom{2n + 1}{n-1-3j}] * q^{n^{2} + n - 3j^{2} - j}
$$
To simplify this expression I set n = floor(n/2 - 1/2)


In [51]:
importlib.reload(analysis_utils)

# find the 
GF = 2
test_list = [i for i in range(2, 20)]
result_list = []
for test in test_list:
    result_list.append(analysis_utils.chol_sqrt_binom_size_growth(GF, test))

# output
print("n", end="    ")
print("num matrices")
for i in range(len(test_list)):
    spacing = " " * (5 - len(str(test_list[i])))
    print(str(test_list[i]) + spacing + str(result_list[i]))



n    num matrices
2    2
3    6
4    28
5    192
6    1920
7    28672
8    618496
9    19132416
10   853508096
11   54584672256
12   5007193669632
13   656391261913088
14   122997864673574912
15   32863344789678981120
16   12522141669213840867328
17   6792468641646224590503936
18   5245768641382970836578729984
19   5760420786247318241425022255104


### Output when run over n of 2 to 10
|n  | num matrices|
|---|-------------| 
|2  |  2          |
|3  |  6          |
|4  |  28         |
|5  |  192        |
|6  |  1920       |
|7  |  28672      | 
|8  |  618496     |
|9  |  19132416   |
|10 |  853508096  | 